client/context/broadcast.go

```diff
@@ -29,6 +31,27 @@ func (ctx CLIContext) BroadcastTx(txBytes []byte) (res sdk.TxResponse, err error
 	return res, err
 }
 
+func CheckTendermintError(err error) *sdk.TxResponse {
+	if err == nil {
+		return nil
+	}
+
+	if err == mempool.ErrTxInCache {
```

Comment:
Why have a unique case for this error and not have it be part of the `switch` clause?

Reply:
Because one is checking for an instance and the switch is checking for types. I'm not sure why the errors are implemented this way?

Reply:
Yikes! I see what you mean. However, with Go 1.13, we can now use the following:

```go
switch {
case xerrors.Is(err, ErrTxInCache):
case xerrors.As(err, &ErrTxTooLarge{}):
// ...
}
```

_Note_: Using `xerrors` instead of the stdlib `errors` gives us backward compat. for go < 1.13

https://github.com/golang/go/wiki/ErrorValueFAQ